internal/logic/logic.go

```diff
@@ -49,5 +63,175 @@ func (l htmlPdfServiceLogic) Ping(req *model.PingRequest) *respModel.Response {
 
 func (l htmlPdfServiceLogic) HealthCheck() bool {
 	// check all internal services are working fine
-	return l.dummyDsSvc.HealthCheck()
+	return l.dsSvc.HealthCheck() && l.htSvc.HealthCheck()
+
+}
+
+func (l htmlPdfServiceLogic) Upload(file io.Reader) *respModel.Response {
+	fileBytes, err := ioutil.ReadAll(file)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrReadFileFail),
+			Data:    nil,
+		}
+	}
+	jb, err := l.htSvc.GetJsonFromHtml(fileBytes)
+	if err != nil {
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFileConversionFail),
+			Data:    nil,
+		}
+	}
+	u := uuid.NewString()
+	err = l.dsSvc.SaveFile(u, jb, 0)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFileStoreFail),
+			Data:    nil,
+		}
+	}
+	return &respModel.Response{
+		Status:  http.StatusCreated,
+		Message: "SUCCESS",
+		Data: map[string]interface{}{
+			"id": u,
+		},
+	}
+}
+
+func (l htmlPdfServiceLogic) Replace(id string, file io.Reader) *respModel.Response {
+	_, err := l.dsSvc.GetFile(id)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrKeyNotFound),
+			Data:    nil,
+		}
+	}
+	fileBytes, err := ioutil.ReadAll(file)
```

Comment:
failure case for this section is pending.

Reply:
done